### PR TITLE
[com_fields] Update the values correctly on save field

### PR DESCRIPTION
**Pull request by laoneo**
Pull Request for Issue #14589.

### Summary of Changes
When the options did change of a list, radio or checkbox field then it didn't work correctly since the move to subformfields in #13069.

### Testing Instructions
- Create a checkbox field with multiple options.
- Edit an article and select some of the options.
- Go back to the field and delete an option you have selected in the article.
- Create a new option.
- Save the field.

### Expected result
The article has still the options selected which are not touched and the option is unselected.

### Actual result
An error is thrown as in #14589.

### CODE REVIEW

File: administrator/components/com_fields/models/field.php [hunk 1]
```diff
@@ -173,6 +177,30 @@ public function save($data)
 	}
 
 	/**
+	 * Convertes the unknown params into an object.
```
Comment (Quy) [inline administrator/components/com_fields/models/field.php]: Typo `Convertes`.

File: administrator/components/com_fields/models/field.php [hunk 2]
```diff
@@ -156,14 +156,18 @@ public function save($data)
...
-			$oldParams = json_decode($field->fieldparams['options']);
-			$newParams = json_decode($data['fieldparams']['options']);
+			$oldParams = $this->getParams($field->fieldparams['options']);
+			$newParams = $this->getParams($data['fieldparams']['options']);
 
-			if (is_object($oldParams) && is_object($newParams) && $oldParams->name != $newParams->name)
+			if (true || is_object($oldParams) && is_object($newParams) && $oldParams != $newParams)
 			{
+				$names = array();
+				foreach ($newParams as $param) {
```
Comment (Quy) [inline administrator/components/com_fields/models/field.php]: When no values specified upon saving: 
`PHP Warning:  Invalid argument supplied for foreach() in C:\xampp\htdocs\joomla-cms-staging\administrator\components\com_fields\models\field.php on line 165`

**Comment by franz-wohlkoenig** [comment]
@laoneo is this PR ready for Test?

**Comment by laoneo** [comment]
@Quy can you retest again, changes some code.

@franz-wohlkoenig let first test @Quy again, then it will be ready for testing.

**Comment by Quy** [comment]
It works except when the value contains a single quote.
`You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 's')' at line 3 `

**Comment by laoneo** [comment]
Thanks for the report, it should be fixed now. Can you retest?

@franz-wohlkoenig it is open for testing now.

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on a483659fcf7ba9d0939a4acd86448ebc7a4d8e53<br /><br />With- and without PR:

- 3 Options, all selected in Article
- 1 Option deleted, 1 new Option appended

In Article 2 Options are selected, the new Option isn't selected.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14681">issues.joomla.org/tracker/joomla-cms/14681</a>.</sub>

**Comment by Quy** [comment]
Now failing with double quotes `"`.
> Fatal error: Call to undefined method FieldsModelField::getParams() in C:\xampp\htdocs\joomla-cms-staging\administrator\components\com_fields\models\field.php on line 159

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on a483659fcf7ba9d0939a4acd86448ebc7a4d8e53<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14681">issues.joomla.org/tracker/joomla-cms/14681</a>.</sub>

**Comment by jeckodevelopment** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14681">issues.joomla.org/tracker/joomla-cms/14681</a>.</sub>

**Pull request merged**